### DISCUSSION TITLE
Does anybody have/had problems with cache size?

### DISCUSSION THREAD

**Discussion by kalemdzievski** [spatie/laravel-permission#1798] [Q&A]
Does anybody have/had problems with the cache size?

We have around 400 permissions and 10 roles. The combinations of roles & permissions result in around ~1500 permissions.

The cache size is whooping 3.8 MB and it's a bit slow to pull it, therefore slowing down every request.
Otherwise, the cache is performing well for other stuff.

We are using:
Version: 3.18.0
Laravel Version: 8.49.0
Cache driver: Redis (The same deal is with the file driver)

**Comment by fredsal** [comment]
Can you make a PR for fix that??

**Comment by erikn69** [comment]
Can you upgrade to `4.2.0` and test again?

**Reply by kalemdzievski** [reply]
@erikn69 just tried. Same result.

**Reply by erikn69** [reply]
@kalemdzievski i made a pull request, try using that branch, and tell me after
Dont forget 
```php
app()->make(\Spatie\Permission\PermissionRegistrar::class)->forgetCachedPermissions()
```
or
```
php artisan permission:cache-reset
```

**Reply by fredsal** [reply]
works for me 👍

**Reply by kalemdzievski** [reply]
@erikn69 I tried it out. I can confirm that this brings the size of the cache for 3.8MB down to ~160kB.
